examples/asset/alter_mesh.rs

```diff
@@ -0,0 +1,200 @@
+//! Shows how to modify mesh assets after spawning.
+
+use bevy::prelude::*;
+use bevy_render::mesh::VertexAttributeValues;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, (setup, instructions))
+        .add_systems(Update, keyboard_controls)
+        .run();
+}
+
+#[derive(Component, Debug)]
+enum Shape {
+    Cube,
+    Sphere,
+}
+
+impl Shape {
+    fn get_model_path(&self) -> String {
+        match self {
+            Shape::Cube => "models/cube/cube.gltf".into(),
+            Shape::Sphere => "models/sphere/sphere.gltf".into(),
+        }
+    }
+
+    fn set_next_variant(&mut self) {
+        *self = match self {
+            Shape::Cube => Shape::Sphere,
+            Shape::Sphere => Shape::Cube,
+        }
+    }
+}
+
+#[derive(Component, Debug)]
+struct Left;
+
+fn setup(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    let left_shape = Shape::Cube;
+    let right_shape = Shape::Cube;
+
+    let left_shape_model = asset_server.load(
+        GltfAssetLabel::Primitive {
+            mesh: 0,
+            primitive: 0,
+        }
+        .from_asset(left_shape.get_model_path()),
+    );
+    let right_shape_model = asset_server.load(
+        GltfAssetLabel::Primitive {
+            mesh: 0,
+            primitive: 0,
+        }
+        .from_asset(right_shape.get_model_path()),
+    );
+
+    // Add a material asset directly to the materials storage
+    let material_handle = materials.add(StandardMaterial {
+        base_color: Color::srgb(0.6, 0.8, 0.6),
+        ..default()
+    });
+
+    commands.spawn((
+        Left,
+        Name::new("Left Shape"),
+        PbrBundle {
+            mesh: left_shape_model,
+            material: material_handle.clone(),
+            transform: Transform::from_xyz(-3.0, 0.0, 0.0),
+            ..default()
+        },
+        left_shape,
+    ));
+
+    commands.spawn((
+        Name::new("Right Shape"),
+        PbrBundle {
+            mesh: right_shape_model,
+            material: material_handle,
+            transform: Transform::from_xyz(3.0, 0.0, 0.0),
+            ..default()
+        },
+        right_shape,
+    ));
+
+    commands.spawn((
+        Name::new("Point Light"),
+        PointLightBundle {
+            transform: Transform::from_xyz(4.0, 5.0, 4.0),
+            ..default()
+        },
+    ));
+
+    commands.spawn((
+        Name::new("Camera"),
+        Camera3dBundle {
+            transform: Transform::from_xyz(0.0, 3.0, 20.0).looking_at(Vec3::ZERO, Vec3::Y),
+            ..default()
+        },
+    ));
+}
+
+fn instructions(mut commands: Commands) {
```

Comment:
This is usually called `spawn_text` or similar in other examples. Can we keep them consistent?

Reply:
Oh, TIL. Can do!